sunbeam-python/sunbeam/plugins/observability/plugin.py

```diff
@@ -423,6 +493,70 @@ def run(self, status: Optional[Status] = None) -> Result:
         return Result(ResultType.COMPLETED)
 
 
+class RemoveGrafanaAgentK8sStep(BaseStep, JujuStepHelper):
+    """Remove Grafana Agent k8s using Terraform"""
+
+    def __init__(
+        self,
+        plugin: "ObservabilityPlugin",
+        tfhelper: TerraformHelper,
+        tfhelper_cos: TerraformHelper,
+        jhelper: JujuHelper,
+    ):
+        super().__init__("Remove Grafana Agent k8s", "Removing Grafana Agent k8s")
+        self.tfhelper = tfhelper
+        self.tfhelper_cos = tfhelper_cos
+        self.jhelper = jhelper
+        self.model = OPENSTACK_MODEL
+        self.read_config = lambda: plugin.get_plugin_info().get(
+            "grafana-agent-k8s-config", {}
+        )
+        self.update_config = lambda c: plugin.update_plugin_info(
+            {"grafana-agent-k8s-config": c}
+        )
+
+    def run(self, status: Optional[Status] = None) -> Result:
+        """Execute configuration using terraform."""
+        app = "grafana-agent-k8s"
+        cos_backend = self.tfhelper_cos.backend
+        cos_backend_config = self.tfhelper_cos.backend_config()
+        try:
+            config = self.read_config()
+        except ConfigItemNotFoundException as e:
+            LOG.exception("Failed removing %s: unable to read config", app)
+            return Result(ResultType.FAILED, str(e))
+
+        tfvars = {
```

Comment:
Is L#529-L537 necessary? Will it make any difference if tfvars are not written?

Reply:
I was referencing the grafana-agent (machine charm) implementation. Let me try to remove everything related to observability model, and the tfvars. I will need to run some manual test

Reply:
Clean up this part a bit more.